assets/js/frontend/add-to-cart-variation.js

```diff
@@ -1,238 +1,218 @@
 /*global wc_add_to_cart_variation_params, wc_cart_fragments_params */
-/*!
- * Variations Plugin
- */
 ;(function ( $, window, document, undefined ) {
+	/**
+	 * Object to handle the variation form.
+	 */
+	var wc_variation_form = {
+		$form               : false,
+		$attributeFields    : false,
+		$singleVariation    : false,
+		$singleVariationWrap: false,
+		$product            : false,
+		$resetVariations    : false,
+		variationData       : false,
+		useAjax             : false,
+		xhr                 : false,
+		/**
+		 * Initialize form handler.
+		 */
+		init: function( $form ) {
+			this.$form                = $form;
+			this.$attributeFields     = $form.find( '.variations select' );
+			this.$singleVariation     = $form.find( '.single_variation' ),
+			this.$singleVariationWrap = $form.find( '.single_variation_wrap' );
+			this.$resetVariations     = $form.find( '.reset_variations' );
+			this.$product             = $form.closest( '.product' );
+			this.variationData        = $form.data( 'product_variations' );
+			this.useAjax              = false === this.variationData;
+
+			// Initial state.
+			this.$singleVariationWrap.show();
+			this.$form.unbind( 'check_variations update_variation_values found_variation' );
+			this.$resetVariations.unbind( 'click' );
+			this.$attributeFields.unbind( 'change ' );
+
+			// Methods.
```

Comment:
Nitpicking but perhaps we can align these. 
Also is binding really needed here?